### PR TITLE
Homework4: Aleksandr Strokov

### PR DESCRIPTION
**Pull request by alstrokov**

### CODE REVIEW

File: src/task1.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+"""task 1
+
+Создайте декоратор, который хранит в себе результаты вызовов функций
+и время получения результата за время запуска программы. """
+
+import time
+
+# global dictionary to sotre time and result of decorated functions runs.
+decorator_run_statistic = dict()
```
Comment (belovmd) [inline src/task1.py:9]: why don't we use a = {}

File: src/task2.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            curr = x
+        else:
+            append_range(rez, start, curr)
+            start = x
+            curr = x
+    append_range(rez, start, curr)
+    return rez
+
+
+print(get_ranges([0, 1, 2, 3, 4, 7, 8, 10]))
```
Comment (belovmd) [inline src/task2.py]: wrong output format - it marked after // in task description
overall it's good solution

**Review by belovmd — COMMENTED** [review]
please address these minor issues

**Comment by alstrokov** [comment]
Ready to be reviewed

File: src/task1.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+# decorator
+def run_and_save_decorator(funct):
+    global decorator_run_statistic
+
+    def wrapper(*args, **kwargs):
+        rez = funct(*args, **kwargs)
+        time_run = time.time()
+        decorator_run_statistic[time_run] = rez
```
Comment (belovmd) [inline src/task1.py:19]: how can I detect which function was called?

File: src/task2.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    rez = list()
+    start = lst[0]
+    curr = lst[0]
+
+    for x in lst:
+        if x - curr <= 1:
+            curr = x
+        else:
+            append_range(rez, start, curr)
```
Comment (belovmd) [inline src/task2.py:37]: it's not best idea to change mutable inside function. we can just list.append(func) for example

**Review by belovmd — APPROVED** [review]
8

**Pull request closed without merging**